### ISSUE TITLE
Setting the default id to other than 1 messes up background datasets

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#943]
An example of the problem:

```
In [6]: ui.clean()

In [7]: ui.set_default_id('bob')

In [8]: ui.load_pha('sherpa-test-data/sherpatest/9774.pi')
read ARF file sherpa-test-data/sherpatest/9774.arf
read RMF file sherpa-test-data/sherpatest/9774.rmf
read background file sherpa-test-data/sherpatest/9774_bg.pi

In [9]: ui.set_bkg_source(ui.xspowerlaw.bpl)

In [10]: ui.get_bkg_source()
Out[10]: <XSpowerlaw model instance 'xspowerlaw.bpl'>

In [11]: ui.fit_bkg()
---------------------------------------------------------------------------
IdentifierErr                             Traceback (most recent call last)
<ipython-input-11-73c81ec708b7> in <module>
----> 1 ui.fit_bkg()

<string> in fit_bkg(id, *otherids, **kwargs)

~/sherpa/sherpa-pr/sherpa/astro/ui/utils.py in fit_bkg(self, id, *otherids, **kwargs)
  10015         """
  10016         kwargs['bkg_only'] = True
> 10017         self._fit(id, *otherids, **kwargs)
  10018
  10019     def _fit(self, id=None, *otherids, **kwargs):

~/sherpa/sherpa-pr/sherpa/astro/ui/utils.py in _fit(self, id, *otherids, **kwargs)
  10035
  10036         if fit_bkg:
> 10037             ids, f = self._get_bkg_fit(id, otherids, numcores=numcores)
  10038         else:
  10039             ids, f = self._get_fit(id, otherids, numcores=numcores)

~/sherpa/sherpa-pr/sherpa/astro/ui/utils.py in _get_bkg_fit(self, id, otherids, estmethod, numcores)
   9839     def _get_bkg_fit(self, id, otherids=(), estmethod=None, numcores=1):
   9840
-> 9841         fit_to_ids, datasets, models = self._prepare_bkg_fit(id, otherids)
   9842
   9843         # Do not add backgrounds to backgrounds.

~/sherpa/sherpa-pr/sherpa/astro/ui/utils.py in _prepare_bkg_fit(self, id, otherids)
   9833         # If no data sets have models assigned to them, stop now.
   9834         if len(models) < 1:
-> 9835             raise IdentifierErr("nomodels")
   9836
   9837         return fit_to_ids, datasets, models

IdentifierErr: model stack is empty
```

Note: `get_bkg_source` succeeded but `fit_bkg` failed.

The problem is that the default id for the background is now `"bob"` according to the UI layer, but the DataPHA object still thinks it is 1

```
In [23]: ui.get_default_id()
Out[23]: 'bob'

In [24]: ui.get_data().default_background_id
Out[24]: 1
```

The easiest solution - taken in #944 (was originally in #931 but moved out) - is for `set_default_id` to not change the background id. We could also change the background id but that would require passing data down to the DataPHA constructor, which I am not sure is worth it at this time.

**Comment by anetasie** [comment]
@DougBurke in CIAO 4.12. 'show_model` shows the background model component in the model expression, but the `fit()` fails.

```
sherpa In [8]: fit()                                                                                        
ModelErr: background model 1 for data set bla has not been set

sherpa In [9]: show_model()                                                                                 
Model: bla
apply_rmf(apply_arf((75141.227687398 * (powlaw1d.p1 + 0.0450573 * (powlaw1d.bb)))))
   Param        Type          Value          Min          Max      Units
   -----        ----          -----          ---          ---      -----
   p1.gamma     thawed            1          -10           10           
   p1.ref       frozen            1 -3.40282e+38  3.40282e+38           
   p1.ampl      thawed            1            0  3.40282e+38           
   bb.gamma     thawed            1          -10           10           
   bb.ref       frozen            1 -3.40282e+38  3.40282e+38           
   bb.ampl      thawed            1            0  3.40282e+38     

sherpa In [10]: show_bkg_model()                                                                            
IdentifierErr: background data set bla in PHA data set bla has not been set
```

**Comment by DougBurke** [comment]
They second one should be fixed by #944 and I would hope the first one is too (although something odd is going on with the error message you show).
